research/utils/visdom_utils.py

```diff
@@ -1,51 +1,61 @@
-import numpy as np
 
 import os
 import time
-import random
+import hydra
+import matplotlib
 
-import cv2
-from visdom import Visdom
-from visdom import server
-from torchvision.transforms import ToTensor
-from torch import Tensor
 import matplotlib.pyplot as plt
-import matplotlib
-import pdb
-import signal
-import hydra
+import numpy as np
+
+from dataclasses import dataclass
+from omegaconf import DictConfig
+from visdom import server, Visdom
 
 
 matplotlib.use("tkagg")
 
 
-@hydra.main(config_path="configs/visdom.yaml")
-def make_visdom(cfg):
+@hydra.main(config_path="../configs", config_name="default.yaml")
+def make_visdom(cfg: DictConfig):
     vdb = Visualizer(cfg)
 
 
+@dataclass
+class VisPlot:
+    var_name: str # Field name in the engine state (i.e. engine.outputs.var_name)
+    plot_key: str # Plot used to plot data. Non-existant plot key creates new plot 
+    split: str # Legend split
+    title: str = None # Title
+    x_label: str = None
+    y_label: str = None
+
+@dataclass
+class VisImg:
+    var_name: str
+    caption: str = None
+    title: str = None
+
+
 class Visualizer:
-    def init(self, cfg):
+    def __init__(self, cfg: DictConfig):
+        self.cfg = cfg.visdom
 
-        self.visdom_config = cfg
         visdom_command = (
-            "screen -S visdom_"
-            + str(visdom_config["port"])
-            + ' -d -m bash -c "python -m visdom.server -port '
-            + str(visdom_config["port"])
-            + '"'
+            'screen -S visdom_{} -d -m bash -c "python -m visdom.server -port {}"'.format(self.cfg.port, self.cfg.port)
         )
+
+        os.mkdir("visdom")
         os.system(visdom_command)
         time.sleep(2)
-        self.env = cfg.env_name
-        self.vis = visdom.Visdom(
-            port=cfg.port,
-            log_to_filename="temp/" + cfg.log_to_filename,
-            offline=cfg.offline,
+        self.env = self.cfg.env_name  # TODO: What is this
```

Comment:
This name lets you separate by pages, so before I had graphs, training images and their predictions for segmentations, and thrid page was evaluation images and their segnmentation

Reply:
Didn't know that, but that functionality is now added. You just specify `env` in the Visdom message as you normally would.